### PR TITLE
Revert "analytics: bail out for detached targets"

### PR DESCRIPTION
**Pull request by samouri**
Reverts ampproject/amphtml#32637

Potential "fix" to https://github.com/ampproject/error-reporting/issues/54.
Setting this up in case we want to eventually merge, we'll have the CI predone.

### CODE REVIEW

**Review by erwinmombay — APPROVED** [review]

**Review by calebcordry — APPROVED** [review]

**Comment by samouri** [comment]
tldr: i messed up.

Did more digging. I was very wrong. Most of the flows _do work_ when the node is detached. Both `element.ownerDocument` and `element.ownerDocument.defaultView` both work on detached nodes.

The bug I was trying to address was this: Screen Shot 2021-02-10 at 11 04 33 AM.

I traced through the mangled names in amp-form and amp-analytics, the failure flow is this: 
1. triggerAnalyticsEvent(...)
2. this.findRoot(target) 
3. getParentWindowFrameElement(target)
4. getTopWindow((target.ownerDocument||target).defaultView)) --> win.__AMP_TOP

Other notes:
- The error we are seeing is that we are trying to read a property on undefined. Working backwards this means that the `.defaultView` is evaluating to undefined.
- When called on a `Document`, defaultView can _never_ be undefined. It is always either `Window|null`. This means that `target.ownerDocument||target` evaluates to a non Document.
- `element.ownerDocument` can only be null if the element is itself a document, which is exactly what this is expression is built for. This means that `target` must not have been a `Node`. 


Since the original case was calling analytics on a `HTMLIframeElement`  thats also impossible to ever happen. Except during the one commit where I was testing out targeting an `AmpDoc` which I reverted before merging 🤕.
